File: clients/typescript-angular/API/Client/SchemaApi.ts
/// <reference path="api.d.ts" />

/* tslint:disable:no-unused-variable member-ordering */

namespace API.Client {
    'use strict';

    export class SchemaApi {
        protected basePath = 'https://localhost/api/v1';
        public defaultHeaders : any = {};

        static $inject: string[] = ['$http', '$httpParamSerializer'];

        constructor(protected $http: ng.IHttpService, protected $httpParamSerializer?: (d: any) => any, basePath?: string) {
            if (basePath) {
                this.basePath = basePath;
            }
        }

        private extendObj<T1,T2>(objA: T1, objB: T2) {
            for(let key in objB){
                if(objB.hasOwnProperty(key)){
                    objA[key] = objB[key];
                }
            }
            return <T1&T2>objA;
        }

        /**
         * Get model schemata for data objects returned by this API.
         * 
         * @param model Optional model filter. If omitted, will return all models.
         */
        public schemaGet (model?: string, extraHttpRequestParams?: any ) : ng.IHttpPromise<inline_response_200_1> {
            const path = this.basePath + '/schema';

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            if (model !== undefined) {
                queryParameters['model'] = model;
            }

            let httpRequestParams: any = {
                method: 'GET',
                url: path,
                json: true,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
        /**
         * Returns help text &amp; subject list for websocket usage.
         * 
         */
        public schemaWebsocketHelp (extraHttpRequestParams?: any ) : ng.IHttpPromise<inline_response_200_1> {
            const path = this.basePath + '/schema/websocketHelp';

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            let httpRequestParams: any = {
                method: 'GET',
                url: path,
                json: true,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
    }
}

File: servers/spring/src/main/java/io/swagger/api/SchemaApi.java
package io.swagger.api;

import io.swagger.model.Error;

import io.swagger.annotations.*;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.SpringCodegen", date = "2016-07-05T09:41:19.947-05:00")

@Api(value = "schema", description = "the schema API")
public interface SchemaApi {

    @ApiOperation(value = "Get model schemata for data objects returned by this API.", notes = "", response = Object.class, tags={ "Schema", })
    @ApiResponses(value = { 
        @ApiResponse(code = 200, message = "Request was successful", response = Object.class),
        @ApiResponse(code = 400, message = "Parameter Error", response = Object.class),
        @ApiResponse(code = 401, message = "Unauthorized", response = Object.class),
        @ApiResponse(code = 404, message = "Not Found", response = Object.class) })
    @RequestMapping(value = "/schema",
        produces = { "application/json", "application/xml", "text/xml", "application/javascript", "text/javascript" }, 
        consumes = { "application/json", "application/x-www-form-urlencoded" },
        method = RequestMethod.GET)
    ResponseEntity<Object> schemaGet(@ApiParam(value = "Optional model filter. If omitted, will return all models.") @RequestParam(value = "model", required = false) String model);


    @ApiOperation(value = "Returns help text & subject list for websocket usage.", notes = "", response = Object.class, tags={ "Schema", })
    @ApiResponses(value = { 
        @ApiResponse(code = 200, message = "Request was successful", response = Object.class),
        @ApiResponse(code = 400, message = "Parameter Error", response = Object.class),
        @ApiResponse(code = 401, message = "Unauthorized", response = Object.class),
        @ApiResponse(code = 404, message = "Not Found", response = Object.class) })
    @RequestMapping(value = "/schema/websocketHelp",
        produces = { "application/json", "application/xml", "text/xml", "application/javascript", "text/javascript" }, 
        consumes = { "application/json", "application/x-www-form-urlencoded" },
        method = RequestMethod.GET)
    ResponseEntity<Object> schemaWebsocketHelp();

}
